neptune/alpha/internal/containers/disk_queue.py

```diff
@@ -61,32 +70,45 @@ def put(self, obj: T) -> None:
         if self._file_size + len(_json) > self._max_file_size:
             self._writer.close()
             self._write_file_idx += 1
-            self._writer = open(self._current_write_log_file(), "a")
+            self._writer = open(self._get_log_file(self._write_file_idx), "a")
             self._file_size = 0
         self._writer.write(_json + "\n")
+        self._last_put_file.write(self._version_getter(obj))
         self._file_size += len(_json) + 1
 
-    def get(self, dry_run=False) -> Optional[T]:
+    def get(self) -> Optional[T]:
+        last_ack_version = self._last_ack_file.read_local()
+        while True:
+            obj = self._get()
+            if obj is None:
+                return None
+            current_version = self._version_getter(obj)
+            if current_version > last_ack_version:
+                if current_version > last_ack_version + 1:
```

Comment:
What about false positives? E.g. data was enqueued for sending but not yet acknowledged?

Reply:
We already talked about it. We agreed that it is fine to redo some operations in such rare case.